lib/sycamore/sycamore/tests/unit/transforms/test_sort.py

```diff
@@ -61,25 +69,36 @@ def test_metadata_document(self):
             MetadataDocument(),
             Document(text_representation="C"),
             MetadataDocument(),
+            Document(text_representation=None),
         ]
 
-        context = sycamore.init()
+        context = sycamore.init(exec_mode=self.exec_mode)
         docset = context.read.document(doc_list)
 
-        with pytest.raises(Exception):
-            sorted_docset = docset.sort(False, "text_representation")
-            sorted_doc_list = sorted_docset.take_all(include_metadata=True)
+        sorted_docset = docset.sort(False, "text_representation")
+        sorted_doc_list = sorted_docset.take_all(include_metadata=True)
+        assert len(sorted_doc_list) == 3
+        assert sorted_doc_list[0].text_representation == "B"
+        assert sorted_doc_list[1].text_representation == "C"
+        assert sorted_doc_list[2].text_representation == "Z"
 
-        # must include default value for docsets with MetadataDocuments
         sorted_docset = docset.sort(False, "text_representation", "A")
         sorted_doc_list = sorted_docset.take_all(include_metadata=True)
 
-        for i in range(len(sorted_doc_list)):
-            if i == 0 or i == 1:
-                assert isinstance(sorted_doc_list[i], MetadataDocument)
-            elif i == 2:
-                assert sorted_doc_list[i].text_representation == "B"
-            elif i == 3:
-                assert sorted_doc_list[i].text_representation == "C"
-            elif i == 4:
-                assert sorted_doc_list[i].text_representation == "Z"
+        for i in range(3):
```

Comment:
can you stick a comment explaining where the 3rd metadata document is coming from